c2rust-transpile/src/translator/atomics.rs

```diff
@@ -226,22 +194,25 @@ impl<'c> Translation<'c> {
                 ptr.and_then(|ptr| {
                     expected.and_then(|expected| {
                         desired.and_then(|desired| {
-                            let weak = static_order(weak);
```

Comment:
What was wrong with this?  It's doing the same thing.

Reply:
`static_order` used to be generic over any `T`, and I changed it to be monomorphic and take `Option<Ordering>` -> `Ordering`. It isn't appropriately named for a function with fully generic bounds, and this was the only call site applying it to something other than `Ordering`.

Reply:
The logic and error message shouldn't be duplicated, though.  If there's a better name for the function, that's fine, but I kept it short since it's an inner/local function and the name is IMO more descriptive than the previous logic.

Reply:
OK, I'll undo the `static_order` change--it isn't really relevant to this PR anyway.

Reply:
Done.